src/components/auth/RegisterStudent.vue

```diff
@@ -529,31 +528,32 @@ const handleFormSubmit = async (isFormValid) => {
 };
 
 const validateCode = async (studentCode, outerIndex = 0) => {
-  if (studentCode && studentCode !== '') {
-    const activationCode = await fetchDocById('activationCodes', studentCode, undefined, 'admin', true, true).catch(
-      (error) => {
-        errors.value = error;
-        dialogMessage.value =
-          'The code does not belong to any organization \n Please enter a valid code or select: \n "I don`t have code "';
-        showErrorDialog();
-        submitted.value = false;
-        return null;
-      },
-    );
-    if (activationCode.orgId && errors.value === '') {
+  // @TODO: Add proper error handling.
+  if (!studentCode || studentCode === '') return;
+
+  try {
+    const activationCode = await fetchDocById('activationCodes', studentCode, undefined, 'admin', true, true);
+
+    if (activationCode.orgId) {
       state.students[outerIndex].orgName = `${_capitalize(activationCode.orgType)} - ${
```

Comment:
Can we guarantee that these properties of `activationCode` i.e. `activationCode.orgType` exist? Should we add fallbacks or optional chaining?

Reply:
I didn't add optional chaining here so that if `activationCode.orgType` doesn't exist, an error is thrown and caught by the `catch` block. There is definitely big room for improvement on the error handling, but, in my opinion, that should be done in a separate PR?